src/wish.cpp

```diff
@@ -396,6 +396,8 @@ void game::wishmonster( const tripoint &p )
             if( spawn != tripoint_min ) {
                 std::vector<tripoint> spawn_points = closest_tripoints_first( cb->group, spawn );
                 for( auto spawn_point : spawn_points ) {
+                    // Set id to 0 to assign new one on spawn
+                    mon.set_id( 0 );
```

Comment:
It is global because it is supposed to unambiguously refer either to one critter or to lack of a critter, never to more than one. It is only tracked within the bubble because enumerating all critters in existence would be a bit hard.

It would be used within the bubble, but would stay assigned even when out of it. For example, so that ID could refer to a mission target monster or that NPCs could remember their targets after being unloaded from map.